Tests/MongoSwiftSyncTests/UnifiedTestRunner/Matching.swift

```diff
@@ -95,10 +95,12 @@ enum MatchableResult {
 
             if case let .subDocument(actualDoc) = self {
                 for k in actualDoc.keys {
-                    try context.withPushedElt(k) {
-                        guard expectedDoc.keys.contains(k) else {
-                            throw NonMatchingError(expected: nil, actual: actualDoc[k], context: context)
-                        }
+                    guard expectedDoc.keys.contains(k) else {
```

Comment:
so I got a failure here in implementing the rest of the operations and I realized that the output was confusing, because the path contained a bunch of elements from the test file e.g. but then ended with a key that was only in the "actual" output, and it wasn't obvious that the problem was we saw the key in "actual". So I updated this to just more explicitly state what the problem is